src/test/java/org/numenta/nupic/encoders/RandomDistributedScalarEncoderTest.java

```diff
@@ -0,0 +1,438 @@
+		rdse.bucketMap.put(midIdx-1, getRangeAsList(4, 9));
+		rdse.bucketMap.put(midIdx, getRangeAsList(5, 10));
+		rdse.bucketMap.put(midIdx+1, getRangeAsList(6, 11));
+		rdse.bucketMap.put(midIdx+2, getRangeAsList(7, 12));
+		rdse.bucketMap.put(midIdx+3, getRangeAsList(8, 13));
+		rdse.minIndex = midIdx-2;
+		rdse.maxIndex = midIdx+3;
+		
+		// TODO Indices must exist
+		
+		// Test some overlaps
+		try {
+			assertEquals("countOverlapIndices didn't work", 5, rdse.countOverlapIndices(midIdx-2,midIdx-2));
+			assertEquals("countOverlapIndices didn't work", 4, rdse.countOverlapIndices(midIdx-1,midIdx-2));
+			assertEquals("countOverlapIndices didn't work", 2, rdse.countOverlapIndices(midIdx+1,midIdx-2));
+			assertEquals("countOverlapIndices didn't work", 0, rdse.countOverlapIndices(midIdx-2,midIdx+3));
+		} catch (IllegalStateException e) {
+			Assert.fail("countOverlapIndices raised an IllegalStateException");
+			e.printStackTrace();
+		}
+	}
+	
+	@Test
+	public void testOverlapOK()
+	{
+		builder = RandomDistributedScalarEncoder.builder()
+				.name("enc")
+				.resolution(1)
+				.w(5)
+				.n(5*20);
+		rdse = builder.build();
+		
+		int midIdx = rdse.getMaxBuckets()/2;
+		
+		rdse.bucketMap.put(midIdx-3, getRangeAsList(4, 9));
+		rdse.bucketMap.put(midIdx-2, getRangeAsList(3, 8));
+		rdse.bucketMap.put(midIdx-1, getRangeAsList(4, 9));
+		rdse.bucketMap.put(midIdx, getRangeAsList(5, 10));
+		rdse.bucketMap.put(midIdx+1, getRangeAsList(6, 11));
+		rdse.bucketMap.put(midIdx+2, getRangeAsList(7, 12));
+		rdse.bucketMap.put(midIdx+3, getRangeAsList(8, 13));
+		rdse.minIndex = midIdx-3;
+		rdse.maxIndex = midIdx+3;
+		
+		try {
+			assertTrue("overlapOK didn't work", rdse.overlapOK(midIdx, midIdx-1));
+			assertTrue("overlapOK didn't work", rdse.overlapOK(midIdx-2, midIdx+3));
+			assertFalse("overlapOK didn't work", rdse.overlapOK(midIdx-3, midIdx-1));
+			
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(100, 50, 0));
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(100, 50, rdse.getMaxOverlap()));
+			assertFalse("overlapOK didn't work for far values", rdse.overlapOK(100, 50, rdse.getMaxOverlap()+1));
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(50, 50, 5));
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(48, 50, 3));
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(46, 50, 1));
+			assertTrue("overlapOK didn't work for far values", rdse.overlapOK(45, 50, rdse.getMaxOverlap()));
+			assertFalse("overlapOK didn't work for far values", rdse.overlapOK(48, 50, 4));
+			assertFalse("overlapOK didn't work for far values", rdse.overlapOK(48, 50, 2));
+			assertFalse("overlapOK didn't work for far values", rdse.overlapOK(46, 50, 2));
+			assertFalse("overlapOK didn't work for far values", rdse.overlapOK(50, 50, 6));
+		} catch (IllegalStateException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
If we don't do anything with the exception, don't catch it. I'd add ``throws`` to the function and let jUnit catch it.